lineax/_operator.py

```diff
@@ -906,6 +906,78 @@ def out_structure(self):
         return self.operator.out_structure()
 
 
+class WoodburyLinearOperator(AbstractLinearOperator, strict=True):
+    """As [`lineax.MatrixLinearOperator`][], but for specifically a matrix
+    with A + U C V structure, such that the Woodbury identity can be used"""
+
+    A: AbstractLinearOperator
+    C: Inexact[Array, " k k"]
+    U: Inexact[Array, " n k"]
+    V: Inexact[Array, " k n"]
+    UCV: Array
+    tags: frozenset[object] = eqx.field(static=True)
+
+    def __init__(
+        self,
+        A: AbstractLinearOperator,
+        C: Inexact[Array, " k k"],
+        U: Inexact[Array, " n k"],
+        V: Inexact[Array, " k n"],
+        tags: Union[object, frozenset[object]] = (),
```

Comment:
I think we probably shouldn't have any tags at all here.

The tags are our kind of ad-hoc way of avoiding needing to create every combination of `{Diagonal,PositiveDefinite,...}{Matrix,PyTree,...}LinearOperator`. Instead we declare as much of the structure as we can via tags.

However in this case we only really have structure that's interesting: the Woodbury structure. We don't really need to combine this with anything else.